pkg/operator/awsTagController.go

```diff
@@ -0,0 +1,324 @@
+		ProxyConfigs:        configInformerFactory.Config().V1().Proxies().Lister(),
+		RegistryConfigs:     regopInformerFactory.Imageregistry().V1().Configs().Lister(),
+		StorageListers: regopclient.StorageListers{
+			Secrets: kubeInformerFactory.Core().V1().Secrets().
+				Lister().Secrets(defaults.ImageRegistryOperatorNamespace),
+			OpenShiftConfig: openshiftConfigKubeInformerFactory.Core().V1().ConfigMaps().
+				Lister().ConfigMaps(defaults.OpenShiftConfigNamespace),
+			OpenShiftConfigManaged: openshiftConfigManagedKubeInformerFactory.Core().V1().ConfigMaps().
+				Lister().ConfigMaps(defaults.OpenShiftConfigManagedNamespace),
+			Infrastructures: infraConfig.Lister(),
+		},
+	}
+
+	_, err := infraConfig.Informer().AddEventHandler(c.eventHandler())
+	if err != nil {
+		return nil, err
+	}
+	c.cachesToSync = append(c.cachesToSync, infraConfig.Informer().HasSynced)
+	return c, nil
+}
+
+// eventHandler is the callback method for handling events from informer
+func (c *AWSTagController) eventHandler() cache.ResourceEventHandler {
+	const workQueueKey = "aws"
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			infra, ok := obj.(*configv1.Infrastructure)
+			if !ok || infra == nil {
+				return
+			}
+			if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.AWS != nil &&
+				len(infra.Status.PlatformStatus.AWS.ResourceTags) != 0 {
+				c.queue.Add(workQueueKey)
+				return
+			}
+		},
+		UpdateFunc: func(prev, cur interface{}) {
+			oldInfra, ok := prev.(*configv1.Infrastructure)
+			if !ok || oldInfra == nil {
+				return
+			}
+			newInfra, ok := cur.(*configv1.Infrastructure)
+			if !ok || newInfra == nil {
+				return
+			}
+			if oldInfra.Status.PlatformStatus != nil && oldInfra.Status.PlatformStatus.AWS != nil &&
+				newInfra.Status.PlatformStatus != nil && newInfra.Status.PlatformStatus.AWS != nil {
+				if !reflect.DeepEqual(oldInfra.Status.PlatformStatus.AWS.ResourceTags, newInfra.Status.PlatformStatus.AWS.ResourceTags) {
+					c.queue.Add(workQueueKey)
+					return
+				}
+			}
+		},
+	}
+}
+
+// Run is the main method for starting the AWS controller
+func (c *AWSTagController) Run(ctx context.Context) {
+	defer k8sruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting AWS Controller")
+	if !cache.WaitForCacheSync(ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	go wait.Until(c.runWorker, time.Second, ctx.Done())
+
+	klog.Infof("Started AWS Controller")
+	<-ctx.Done()
+	klog.Infof("Shutting down AWS Controller")
+}
+
+func (c *AWSTagController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+// processNextWorkItem is for prcessing the event received
+// which blocks until a new item is received
+func (c *AWSTagController) processNextWorkItem() bool {
+	obj, shutdown := c.queue.Get()
+	if shutdown {
+		return false
+	}
+	defer c.queue.Done(obj)
+
+	klog.V(5).Infof("AWSController: got event from workqueue")
```

Comment:
Here and everywhere in this file:

```suggestion
	klog.V(5).Infof("AWSTagController: got event from workqueue")
```